### PR TITLE
Modulecog session

### PR DESCRIPTION
**Pull request by Fripe070**
### Checklist
- [ ] I have planned and discussed this feature with other contributors.
- [x] I have run basic tests on my code to ensure it is functional.
- [ ] I have thoroughly tested my code with various states, contexts and edge cases.
- [x] I have searched for a similar pull request in this repository and didn't find anything relevant.

### Summary
Adds a `ModuleSessionCog` class to the helpers which inherits from `breadcord.module.ModuleCog` and provides a `session` attribute holding an `aiohttp.ClientSession`.

### Issues
(No related issues)

### Extra information
This was done since interacting with an external API is an extreemely common thing for a module to need to do, and this should remove some boilerplate.

### CODE REVIEW

**Comment by Sophon96** [comment]
imo the `ModuleSessionCog` name is a bit confusing given that it isn't clear that "session" refers to aiohttp.ClientSession.

**Comment by Fripe070** [comment]
Do you have any suggestions for a better name? I agree that it's a bit unclear

**Review by Sophon96 — COMMENTED** [review]
LGTM!!!!

**Comment by Sophon96** [comment]
> Do you have any suggestions for a better name? I agree that it's a bit unclear

Perhaps `HttpModuleCog`?

**Comment by m31-galaxy** [comment]
I like HttpModuleCog more than ModuleSessionCog.

**Comment by Fripe070** [comment]
According to [PEP 8](https://peps.python.org/pep-0008/#descriptive-naming-styles) acronyms in CapWords names should be all uppercase, so `HTTPModuleCog` has been used instead of `HttpModuleCog`
> Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

**Comment by m31-galaxy** [comment]
So be it, then.

File: breadcord/helpers.py [hunk 1]
```diff
@@ -207,18 +209,19 @@ def decorator(cls: type[_Transformer]) -> _Transformer:
     return decorator
 
 
-class HTTPModuleCog(breadcord.module.ModuleCog):
+class HTTPModuleCog(ModuleCog):
```
Comment (Fripe070) [inline breadcord/helpers.py:211]: Changed becasue I was getting some weird recursive import error, not sure why, didn't happen before, but I would prefer for it not to happen in production

**Review by m31-galaxy — APPROVED** [review]
